### PR TITLE
NRPT-549 Fix Core importer not updaing collections and records

### PR DESCRIPTION
**Pull request by BcGovNeal**
https://bcmines.atlassian.net/browse/NRPT-549

- Added `20201117185607-addCoreCollectionRefId.js` migration to associate collections with the `permit_amendment_guid` from Core using the existing `_sourceRefId` on permit records
- Refactored Collection controller so the update function can be called directly from another file
- Added logic to update collections and records if changes are detected

Make sure to run the migration first if testing locally

### CODE REVIEW

File: api/src/controllers/collection-controller.js [hunk 1]
```diff
@@ -64,8 +61,26 @@ exports.protectedPut = async function (args, res, next) {
     defaultLog.info(`protectedPut - you must provide an object`);
     queryActions.sendResponse(res, 400, {});
     next();
+  }  
+
+  let obj = null;
+  try {
+    // obj = await Put.updateById(collectionId, updateObj);
```
Comment (mtCarto) [inline api/src/controllers/collection-controller.js:68]: remove this commented line

**Review by mtCarto — APPROVED** [review]
Collection is updated appropriately 👍

File: api/src/controllers/collection-controller.js (hunk 1 above)
Comment (BcGovNeal) [reply api/src/controllers/collection-controller.js]: Fixed in https://github.com/bcgov/NRPTI/pull/690/commits/2bd4d09487e3927447fa6e30c778887452e5a60b

**Review by MaxWardle — APPROVED** [review]
Migration runs fine

**Pull request merged**